src/http/states/writeHeaderLines/WriteHeaderLines.cpp

```diff
@@ -25,18 +32,43 @@ WriteHeaderLines::WriteHeaderLines(Client* context)
 }
 
 void WriteHeaderLines::run()
-{
+try {
   Headers& headers = _client->getResponse().getHeaders();
-  headers.addHeader("Date", "Wed, 29 Oct 2025 12:00:00 GMT"); // TODO
-  headers.addHeader("Server", "webserv"); // TODO from config probaly
+  headers.addHeader("Date", _makeHttpDate());
+  headers.addHeader("Server", "webserv"); // TODO from config probably ?
   headers.addHeader("Connection", "close");
 
   IBuffer& outBuffer = _client->getOutBuff();
   outBuffer.append(headers.toString());
   outBuffer.append(http::CRLF);
 
   _client->getStateHandler().setState<WriteBody>();
+} catch (const std::runtime_error& e) {
+  _log.error() << "WriteHeaderLines: " << e.what() << "\n";
+  _client->getResponse().setStatusCode(StatusCode::InternalServerError);
+  throw e; // todo catch me if you can ;)
```

Comment:
Always rethrow like this:
```suggestion
  throw; // todo catch me if you can ;)
```
Otherwise a copy gets rethrown, not the same object. More importantly, it would throw the type of the catched reference, not the actual derived type.

Reply:
yes i noticed that